example/lib/calculator/service_handler.ex

```diff
@@ -18,13 +22,34 @@ defmodule Calculator.ServiceHandler do
   end
 
   @impl true
+  def divide(_left, 0) do
+    raise Calculator.Generated.DivideByZeroError, message: "Cannot divide by zero"
+  end
+
   def divide(left, right) do
-    case right do
-      0 ->
-        raise Calculator.Generated.DivideByZeroError, message: "Cannot divide by zero"
+    div(left, right)
+  end
+
+  @impl true
+  def vector_product(left, right, type) do
```

Comment:
Could also match on `VectorProductType`. 😉

Reply:
This is more tricky since it's a macro. First approach would be
```elixir
def vector_product(left, right, VectorProductType.dot_product())
```
but the compiler will not allow you to do this.

Second approach is:
```elixir
@dot_product VectorProductType.dot_product()

def vector_product(left, right, @dot_product)
```
But this is just plain ugly.
I settled with the final approach here.

Reply:
This is a very good example of why #312 would be useful btw

Reply:
Oh, that's right. I was thinking we had already done something with #313.